src/transformers/generation/utils.py

```diff
@@ -4562,13 +4668,7 @@ def _assisted_decoding(
                 candidate_logits = candidate_logits.to(self.device)
 
             candidate_length = candidate_input_ids.shape[1] - input_ids.shape[1]
-            last_assistant_token_is_eos = (
-                ~candidate_input_ids[:, -1]
-                .tile(eos_token_id_tensor.shape[0], 1)
-                .ne(eos_token_id_tensor.unsqueeze(1))
-                .prod(dim=0)
-                .bool()
-            )
+            last_assistant_token_is_eos = stopping_criteria[-1](candidate_input_ids, None)
```

Comment:
this means we the last stopping_criteria is always `EOSTokenCriteria`. It's not necessarily obvious and if people use custom passed criteria I am not sure this will always be the case?

Reply:
right, will fix it

Reply:
@ArthurZucker good catch, I missed this one too :)

Reply:
@gante need you to look. I removed some code, given that assisted decoding works only with one batch, so we can make some assumptions about when to stop generating.